android/src/main/java/app/notifee/core/AlarmPermissionBroadcastReceiver.java

```diff
@@ -0,0 +1,19 @@
+package app.notifee.core;
+
+import static android.app.AlarmManager.ACTION_SCHEDULE_EXACT_ALARM_PERMISSION_STATE_CHANGED;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+
+public class AlarmPermissionBroadcastReceiver extends BroadcastReceiver {
+  @Override
+  public void onReceive(Context context, Intent intent) {
+
+    if (intent.getAction().equals(ACTION_SCHEDULE_EXACT_ALARM_PERMISSION_STATE_CHANGED)) {
+      Log.i("AlarmPermissionReceiver", "Received alarm permission state changed event");
+      new NotifeeAlarmManager().rescheduleNotifications();
```

Comment:
Keep this as its own `BroadcastReceiver` sounds better, as I feel it'll give more flexibility in the future